client/extensions/woocommerce/app/dashboard/index.js

```diff
@@ -50,49 +50,61 @@ class Dashboard extends Component {
 		className: PropTypes.string,
 		finishedInitialSetup: PropTypes.bool,
 		hasOrders: PropTypes.bool,
+		isSetupComplete: PropTypes.bool,
 		loading: PropTypes.bool,
 		selectedSite: PropTypes.shape( {
 			ID: PropTypes.number.isRequired,
 			slug: PropTypes.string.isRequired,
 			URL: PropTypes.string.isRequired,
 		} ),
+		siteId: PropTypes.number,
 		mailChimpConfigured: PropTypes.bool,
 		fetchOrders: PropTypes.func,
 		fetchSetupChoices: PropTypes.func,
 		requestSyncStatus: PropTypes.func,
+		setupChoicesLoading: PropTypes.bool,
 	};
 
 	componentDidMount = () => {
-		const { selectedSite, productsLoaded } = this.props;
+		const { siteId, productsLoaded } = this.props;
 
-		if ( selectedSite && selectedSite.ID ) {
-			this.props.fetchSetupChoices( selectedSite.ID );
-			this.props.fetchOrders( selectedSite.ID );
-			this.props.requestSettings( selectedSite.ID );
+		if ( siteId ) {
```

Comment:
I think that this could be extracted. Fetching here a and in componentWillReceiveProps is the same.

Reply:
I agree with @budzanowski -- it would be nice to extract this into a method. And instead of `componentWillReceiveProps`, we can use `componentDidUpdate`. The benefit is that we can use `this.props` in both cases with the same meaning.